src/Composer/ScriptHandler.php

```diff
@@ -42,6 +42,28 @@ public static function generateSymlinks(Event $event)
     }
 
     /**
+     * Sets environment variable for random secret on installation.
+     *
+     * @param Event $event The event object
+     */
+    public static function generateRandomSecret(Event $event)
+    {
+        $extra = $event->getComposer()->getPackage()->getExtra();
+
+        if (!isset($extra['incenteev-parameters']) || !isset($extra['incenteev-parameters']['file'])) {
+            return;
+        }
+
+        $file = $extra['incenteev-parameters']['file'];
+
+        if (is_file($file)) {
+            return;
+        }
+
+        putenv('CONTAO_RANDOM_SECRET=' . md5(uniqid(mt_rand(), true)));
```

Comment:
you can never change the environment of a parent process. So this environment variable will only be accessible to the current process (running Composer) and subprocesses launched from this process (subsequent composer scripts, or ``git`` commands used by the installer)